pkg/controller/bastion/options.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net"
+
+	openstackapi "github.com/gardener/gardener-extension-provider-openstack/pkg/apis/openstack"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/security/rules"
+
+	"github.com/gardener/gardener/extensions/pkg/controller"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+//Maximum length for "base" name due to fact that we use this name to name other openstack resources,
+//
+const (
+	maxLengthForBaseName = 33
+	imageRef             = "13850abb-e451-45a2-8437-bdf6de530a19" //m1.tiny
```

Comment:
This seems to be a SAP internal assumption.

Reply:
Yes it is SAP internal image

Reply:
Any plans to change that? A hard coded SAP image id should not stay in the code.

Reply:
ticket opened in https://github.com/gardener/gardener-extension-provider-openstack/issues/395

Reply:
some update for image and flavor in 7719f6ef

Reply:
As written in our private chat I would recommend to make these settings configurable via the extension config.

Reply:
Hi @dkistner I'm moving these specifics to bastion type now, it seems less effort.  a6cdad6d. let me know if any comments.

Reply:
no, this should not be part of the bastion resource

Reply:
> no, this should not be part of the bastion resource
ok I will try to move to extension config, btw, is any anything I should be aware of how to identify either belong to bastion resource or extension config.

Reply:
@dkistner add extension config part 7e0fdff5 per your comment. Please have a look when you have time. and let me know if I missing something else . Thanks.